central/networkgraph/flow/datastore/flow_impl.go

```diff
@@ -131,3 +132,16 @@ func (fds *flowDataStoreImpl) RemoveMatchingFlows(ctx context.Context, keyMatchF
 	}
 	return fds.storage.RemoveMatchingFlows(ctx, keyMatchFn, valueMatchFn)
 }
+
+// RemoveStaleFlows - remove stale duplicate network flows
+func (fds *flowDataStoreImpl) RemoveStaleFlows(ctx context.Context) error {
+	if ok, err := networkGraphSAC.WriteAllowed(ctx); err != nil {
+		return err
+	} else if !ok {
+		return sac.ErrResourceAccessDenied
+	}
+	if env.PostgresDatastoreEnabled.BooleanSetting() {
```

Comment:
Seeing this now but I feel the env check should be performed before SAC check to avoid eventual noop.